src/Cake.Common.Tests/Unit/Tools/GitLink/GitlinkRunnerTests.cs

```diff
@@ -0,0 +1,241 @@
+﻿using Cake.Common.Tests.Fixtures.Tools;
+using Cake.Core;
+using Cake.Core.IO;
+using Cake.Testing;
+using Xunit;
+
+namespace Cake.Common.Tests.Unit.Tools.GitLink
+{
+    using System;
+
+    public sealed class GitlinkRunnerTests
+    {
+        public sealed class TheRunMethod
+        {
+            [Fact]
+            public void Should_Throw_If_SolutionPath_Is_Null()
+            {
+                // Given
+                var fixture = new GitLinkFixture();
+                fixture.SolutionPath = null;
+
+                // When
+                var result = Record.Exception(() => fixture.Run());
+
+                // Then
+                Assert.IsArgumentNullException(result, "solutionPath");
+            }
+
+            [Fact]
+            public void Should_Find_GitLink_Runner()
+            {
+                // Given
+                var fixture = new GitLinkFixture();
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal("/Working/tools/gitlink.exe", result.Path.FullPath);
+            }
+
+            [Fact]
+            public void Should_Throw_If_Process_Was_Not_Started()
+            {
+                // Given
+                var fixture = new GitLinkFixture();
+                fixture.GivenProcessCannotStart();
+
+                // When
+                var result = Record.Exception(() => fixture.Run());
+
+                // Then
+                Assert.IsType<CakeException>(result);
+                Assert.Equal("GitLink: Process was not started.", result.Message);
+            }
+
+            [Fact]
+            public void Should_Throw_If_Has_A_Non_Zero_Exit_Code()
+            {
+                // Given
+                var fixture = new GitLinkFixture();
+                fixture.GivenProcessExitsWithCode(1);
+
+                // When
+                var result = Record.Exception(() => fixture.Run());
+
+                // Then
+                Assert.IsType<CakeException>(result);
+                Assert.Equal("GitLink: Process returned an error.", result.Message);
+            }
+
+            [Fact]
+            public void Should_Use_Provided_SolutionPath_In_Process_Arguments()
+            {
+                // Given
+                var fixture = new GitLinkFixture();
+                fixture.SolutionPath = "source";
```

Comment:
I might have chosen a wrong name in GitLink for this. It's actually the repository root on disk, has nothing to do with the solution.